pupil_src/shared_modules/background_helper.py

```diff
@@ -112,28 +122,28 @@ def __del__(self):
 
 
 class IPC_Logging_Task_Proxy(Task_Proxy):
-    def __init__(self, ipc_push_url, name, generator, args=(), kwargs={}):
-        extended_args = [ipc_push_url]
-        extended_args.extend(args)
-        super().__init__(name, generator, args=extended_args, kwargs=kwargs)
-
-    def _wrapper(
-        self, pipe, _should_terminate_flag, generator, ipc_push_url, *args, **kwargs
-    ):
-        self._enforce_IPC_logging(ipc_push_url)
+    push_url = None
+
+    def _prepare_wrapper_args(self, *args):
+        return [*args, self.push_url]
+
+    def _wrapper(self, pipe, _should_terminate_flag, generator, push_url, *args, **kwargs):
+        self.push_url = push_url
         super()._wrapper(pipe, _should_terminate_flag, generator, *args, **kwargs)
 
-    def _enforce_IPC_logging(self, ipc_push_url):
+    def _change_logging_behavior(self):
         """
         ZMQ_handler sockets from the foreground thread are broken in the background.
         Solution: Remove all potential broken handlers and replace by new oneself.
 
-        Caveat: If a broken handler is present is incosistent across environments.
+        Caveat: If a broken handler is present it is incosistent across environments.
         """
+        assert self.push_url, "`push_url` was not set by foreground process"
         del logger.root.handlers[:]
         zmq_ctx = zmq.Context()
-        handler = zmq_tools.ZMQ_handler(zmq_ctx, ipc_push_url)
+        handler = zmq_tools.ZMQ_handler(zmq_ctx, self.push_url)
         logger.root.addHandler(handler)
+        logger.root.setLevel(logging.NOTSET)
```

Comment:
Default `logger.root.level` is set to `logging.WARNING`. This filters logs before sending them onto the IPC. There is already filtering happening on the log-receiving side.